File: Quiz-server/src/main/java/com/qucat/quiz/repositories/dto/WebsocketEvent.java
package com.qucat.quiz.repositories.dto;

import com.qucat.quiz.repositories.dto.game.Users;
import com.qucat.quiz.repositories.entities.Question;
import lombok.Builder;
import lombok.Data;

import java.util.List;

@Data
@Builder
public class WebsocketEvent {
    private EventType type;
    private List<String> players;
    private Question question;
    private Users gameResults;
    private int currQuestion;

    public enum EventType {
        RESULTS,
        QUESTION,
        PLAYERS
    }

}

File: Quiz-web/src/app/modules/core/models/websocketEvent.ts
import {Question} from "./question";
import {Users} from "./gameResults";
import {NotificationDto} from "./notificationDto";
import {Message} from "./message";

export class WebsocketEvent {
  players: String[];
  question: Question;
  gameResults: Users;
  message: Message;
  type:EventType;
  notification: NotificationDto;
}

export enum EventType {
  RESULTS = "RESULTS",
  QUESTION = "QUESTION",
  PLAYERS = "PLAYERS",
  MESSAGE = "MESSAGE",
  NOTIFICATION = "NOTIFICATION"
}
